### ISSUE TITLE
Feature Request: Buttons & Button Set

### ISSUE THREAD

**Issue by damienwebdev** [graycoreio/daffodil#59]
## I'm submitting a...

<pre><code>
[x] Feature request
</code></pre>

## Current behavior
Currently, the button styles and design are littered over the `foundation-demo` codebase. Let's refactor to a singular button design. 

## Expected behavior
A singlular `daff-button` component that can be applied to a `button`, `a` an element with a `aria-role="button`" that applies consistent styling, animations, and interactions. There are two obvious use-cases:

1. Text type buttons
2. Icon type buttons

## Minimal reproduction of the problem with instructions
We would need test cases to cover the following components.

Text Type
```
<a daff-button>Some Text</a>
<button daff-button>Some Text</button>
<div role="button" daff-button>Some Text</div>
```

Icon Type
```
<a daff-button><i class="fa fa-circle"></i></a>
<button daff-button><i class="fa fa-circle"></i></button>
<div role="button" daff-button><i class="fa fa-circle"></i></div>
```

## What is the motivation / use case for changing the behavior?
Design consistency and code maintenance.

**Comment by damienwebdev** [comment]
This is partially completed by #56 but we need to centralize the button styles into a single place.

**Comment by damienwebdev** [comment]
@xelaint would you be able to pickup this PR?

**Comment by xelaint** [comment]
@damienwebdev Yes, I can.

**Comment by damienwebdev** [comment]
Closing as this is resolved by #70
